### ISSUE TITLE
[Proposal] Use RST and convert images generated from LaTeX to use MathJax

### ISSUE THREAD

**Issue by akohlmey** [lammps/lammps#1334]
**Summary**

In case we adopt the changes from #1320, the biggest hurdle for switching to using the RST file format as source for the documentation has been removed. This would be further simplified and more consistent, if we take all equations in latex format and integrate them as MathJax. This would also allow us to use LaTeX typesetting to refer to parameters from those equations in the text.

**Detailed Description**

The markup used by txt2html is simple and has served well as a tool to have a single source for web-based and printable PDF documentation. However, the switch to RST based html generation via Sphinx has significantly improved the visual appeal and accessibility of the online documentation. But since we are still using the .txt files and convert them to RST with a script, we cannot take full advantage of RST and are limited by some of the "hacks" that were used to get desirable typesetting through txt2html, which would be done differently in plain RST. With the added complexity in the documentation, creating a reasonably well looking and typeset PDF file of the documentation has become increasingly difficult.

Thus one possible path to move forward is the following:
- abandon the txt2html processing and using .txt markup for the documentation
- instead do one complete conversion and instead import the resulting .rst files and use those as source for the future documentation.
- incrementally replace all generated images of equations typeset with LaTeX with inline MathJax. once completed, that should simplify maintenance of the documentation and generation of html and pdf manual
- incrementally review the RST markup and replace constructs resulting from txt2html hacks with cleaner RST based formatting
- provide an archive of the latest updated .txt files with the corresponding .html files and generated .pdf files as "Legacy Documentation".

_This most certainly will need to be discussed in person, but we can use this issue as the document where we review and rewrite the proposed steps and ultimately chosen approach. So both, comments below as well as edits to this text are welcome_

**Comment by eofetis** [comment]
If you decide to go the RST way, by simply replacing *.txt with slightly modified *.rst in doc/src and slightly changing Makefile I managed to get nice LaTeX equations in both html (through MathJax) and pdf. Here is comparison with old "jpg style":
1) pdf
![screen shot 2019-02-14 at 3 42 20 pm](https://user-images.githubusercontent.com/42049238/52824939-9de45200-306f-11e9-920e-cb138a2761d8.png)

2) html
![screen shot 2019-02-14 at 3 42 32 pm](https://user-images.githubusercontent.com/42049238/52824951-a63c8d00-306f-11e9-89d6-460b197f85c7.png)

**Comment by rbberger** [comment]
As the author of the `txt2rst` tool (which was always meant to be a gateway drug), I welcome making it obsolete. :smile:

**Comment by eagunn** [comment]
I am volunteering to work on this issue. Based on advice from @akohlmey, I've forked the repo and am working on creating a demo branch that:
- captures the converted rst files
- demonstrates the use of mathjax formatting by substituting inline equations in the rst files for a small subset of the current jpg files
- includes a draft set of changes to the makefile to support building straight from the native rst files and eliminates the use of the current .txt files for everything but the "old" build.

When I have something ready for review, I will create a draft pull request that will allow for the group to weigh in on build choices, etc. I do understand that no final decision to switch over to rst-native documentation has been made. But Axel says a working-demo of a new build process might facilitate the face-to-face discussions in August.

Frankly, most of the heavy lifting for this change has already been completed with the excellent txt2rst tool written by @rbberger . I think the hard choices, which will require the input of the core team, will be deciding whether and how to continue to support the "old" build option. 

However, there are quite a few small changes to the makefile and scripts such as doc_anchor_check.py that need to be made, which seems to me to make this a good project for a newbie dev. I'm traveling this week and only making slow progress. I would not expect to have a draft pull ready till sometime next week.

But I find I may need to start asking some rather detailed questions about tools such as doc_anchor_check.py. I thought I should put my hand up here, first, so you'd know where the questions are coming from.

**Comment by akohlmey** [comment]
@eagunn thanks for volunteering. this is much appreciated. i have sent you a collaborator invite for the LAMMPS GitHub project, so we can assign this issue to you.

no worries about the "old" documentation build options. if we switch to RST markup, the old .txt files will be removed, so the path to the legacy documentation build will be gone.

i expect the major amount of work to be taking advantage of the more flexible typesetting options in RST versus the minimal legacy system that was limited to standard HTML tags without style sheets and related formatting options. that made for some awkward choices, that looked ok in the PDF manual, but were not very logical in terms of the typesetting markup.

**Comment by eagunn** [comment]
@akohlmey Agreed about the major work. I understand implementing this proposal as an enabling change that then opens up possibilities for improving the docs.
And thanks for the hint on the "old". That makes my work simpler. I'll try to produce a clean file tree and make process that omits the txt files and the old build. That will be easier to test and review. Then, we can decide where the archive mentioned in the issue should live and how it should be organized.

**Comment by rbberger** [comment]
@eagunn We've now changed the format to RST. All .rst files are in the src/ folder. Old txt files are in a txt/ folder. Instead of keeping both files around, we are now gradually removing the txt files once we touch the rst files.

The changes I haven't migrated yet are the ones that have updated math equations. As far as I understand, you wrote a tool to do the conversion. Do you have a list of files that you converted with this tool and/or are there some files that are particularly problematic? Are they in a special folder in your PR #1533?

**Comment by eagunn** [comment]
@rbberger  I tried to make it easy to review the conversion and, eventually, correct problem equations. But there are a lot of files and it probably needs to be done file by file. If you look at my pull request or if you go to my repo on github and look at branch iss1334proofOfConcept, you will find the two folders described below. A directory list will give you a list of files. A diff will allow you to see what changes were made:

From my pull request:
_src/modifiedRst/*.rst files: I wrote new script (doc/utils/converters/lammpsdoc/eqImg2mathjaxInline.py) to find image links in the rst files for images from the Eqs folder and substitute math markup straight out of the .tex file for each .jpg. Rather than slap the new versions down into the src folder wholesale, I thought it would be easier to review this conversion if each reviewer picked a few of the modified files for documentation for which they are familiar, moved those into src, rebuilt, and then could review only those selected pages in the output.

src/safeRst/*.rst: The script above makes a copy of the original rst file in this folder. This is a bit redundant but makes it simple to diff the two versions._

I did not have any idea how to make general fixes to the tex markup so I didn't start on that.
